pkg/controller/daemon/daemoncontroller_test.go

```diff
@@ -269,6 +274,22 @@ func validateSyncDaemonSets(t *testing.T, fakePodControl *fakePodControl, expect
 	if len(fakePodControl.DeletePodName) != expectedDeletes {
 		t.Errorf("Unexpected number of deletes.  Expected %d, saw %d\n", expectedDeletes, len(fakePodControl.DeletePodName))
 	}
+	// Every Pod created should have a ControllerRef.
+	if got, want := len(fakePodControl.ControllerRefs), expectedCreates; got != want {
+		t.Errorf("len(ControllerRefs) = %v, want %v", got, want)
+	}
+	// Make sure the ControllerRefs are correct.
+	for _, controllerRef := range fakePodControl.ControllerRefs {
+		if got, want := controllerRef.APIVersion, "extensions/v1beta1"; got != want {
+			t.Errorf("controllerRef.APIVersion = %q, want %q", got, want)
+		}
+		if got, want := controllerRef.Kind, "DaemonSet"; got != want {
```

Comment:
In general, it's good practice to make unit test expected values hard-coded whenever possible. This helps to avoid problems being hidden due to tests that are actually tautologies: "check that A == B but actually both come from C, so the test passes even if C is wrong."

In particular, having hard-coded values in this test saved me in the CronJob PR when I rebased and someone had renamed a package on import. That actually made `controllerKind` use the wrong `SchemeGroupVersion`, but this test caught it.